src/coreclr/vm/mono/mono_coreclr.cpp

```diff
@@ -1529,15 +1523,84 @@ extern "C" EXPORT_API const char* EXPORT_CC mono_image_get_name(MonoImage* image
 
 extern "C" EXPORT_API const MonoTableInfo* EXPORT_CC mono_image_get_table_info (MonoImage * image, int table_id)
 {
+    // TODO: UNUSED - Remove
     ASSERT_NOT_IMPLEMENTED;
     return NULL;
 }
 
-// Generated by UnityEmbedHost.Generator - Commit these changes
-extern "C" EXPORT_API int32_t EXPORT_CC mono_image_get_table_rows(MonoImage* image, int32_t table_id)
+static ASSEMBLYREF ASSEMBLYREF_From_AssemblyIntPtrHandle(MonoImage* assemblyIntPtrHandle)
 {
-    GCX_PREEMP(); // temporary until we sort out our GC thread model
-    return g_HostStruct->image_get_table_rows(image, table_id);
+    OBJECTHANDLE assemblyHandle = handle_from_uintptr((uintptr_t)assemblyIntPtrHandle);
+    OBJECTREF assemblyObjectRef = ObjectFromHandle(assemblyHandle);
+    return ASSEMBLYREF(assemblyObjectRef);
+}
+
+extern "C" EXPORT_API int32_t EXPORT_CC mono_image_get_table_rows(MonoImage* assemblyIntPtrHandle, int32_t table_token)
```

Comment:
are we ok with using .net runtime token here and not mono ones?
the method is prefixed with `mono_` but I don't really want to do a redundant conversions from mono enums given that we move away from mono